### PR TITLE
Feature/remote url

### PR DESCRIPTION
**Pull request by AHReccese**
#### Reference Issues/PRs
#95

### CODE REVIEW

**Comment by AHReccese** [comment]
@sadrasabouri 
kind reminder for this!

File: pymilo/pymilo_obj.py [hunk 1]
```diff
@@ -84,25 +83,29 @@ class Import:
...
+    def __init__(self, file_adr=None, json_dump=None, url=None):
         """
         Initialize the Pymilo Import instance.
 
         :param file_adr: the file path where the serialized model's JSON file is located.
-        :type file_adr: string
+        :type file_adr: str or None
         :param json_dump: the json dump of the associated model, it can be None(reading from the file_adr)
         :type json_dump: str or None
+        :param url: the url in which the previously exported JSON file has been uploaded
```
Comment (sadrasabouri) [inline pymilo/pymilo_obj.py:94]: `url to exported JSON file`

File: pymilo/pymilo_obj.py [hunk 2]
```diff
@@ -84,25 +83,29 @@ class Import:
...
         serialized_model_obj = None
+        if url is not None:
+            serialized_model_obj = download_model(url)
+        elif json_dump is not None and isinstance(json_dump, str):
+            serialized_model_obj = json.loads(json_dump)
+        elif file_adr is not None:
+            with open(file_adr, 'r') as fp:
+                serialized_model_obj = json.load(fp)
+        else:
+            raise Exception("Invalid input parameters, you should either pass a valid file_adr or a json_dump or a url to initiate Import class.")
```
Comment (sadrasabouri) [inline pymilo/pymilo_obj.py:107]: import this error message from `params` instead of hard-coding it, here.

File: pymilo/utils/util.py [hunk 3]
```diff
@@ -135,3 +136,31 @@ def prefix_list(list1, list2):
     if len(list1) < len(list2):
         return False
     return all(list1[j] == list2[j] for j in range(len(list2)))
+
+
+def download_model(url):
+    """
+    Download the model from the given url.
+
+    :param url: the url in which the previously exported JSON file has been uploaded
```
Comment (sadrasabouri) [inline pymilo/utils/util.py:145]: same comment.

File: pymilo/utils/util.py [hunk 4]
```diff
@@ -135,3 +136,31 @@ def prefix_list(list1, list2):
...
+    s.mount('https://', requests.adapters.HTTPAdapter(max_retries=retries))
+    try:
+        response = s.get(url)
+    except Exception:
+        raise Exception("Failed to download the JSON file, Server didn't respond." )
+    try:
+        if response.status_code == 200:
+            return response.json()
+    except ValueError:
+        raise Exception("The downloaded content is not a valid JSON file.")
```
Comment (sadrasabouri) [inline pymilo/utils/util.py:166]: same comment.

**Review by sadrasabouri — APPROVED** [review]

**Pull request merged**